### PR TITLE
Genesys New Queue Performance Method

### PR DESCRIPTION
**Pull request by Diego-H-S**
## Summary
In order to be more general when calling Genesys API, we have changed the method `genesys_generate_exports` and renamed it as `genesys_api_connection` introducing a new parameter `end_point`. Also, a `method` parameter has been added to choose between wither connections, `POST` or `GET`.
It has been removed old methods never used in production but as a testing concept in the very early beginning of the actual project.
Furthermore, some parts of the calls have been modified, adding some time sleep step, to not overwhelm Genesys API, otherwise, we get fewer responses than calls made. 

## Importance
Required by the client to retrieve all queues from Genesys at once. The actual paradigm creates one call per queue.

## Checklist

This PR:
- [x] follows the guidelines laid out in `CONTRIBUTING.md`
- [x] links relevant issue(s)
- [x] adds/updates tests (if appropriate)
- [x] adds/updates docstrings (if appropriate)
- [x] adds an entry in `CHANGELOG.md`

### CODE REVIEW

File: viadot/sources/genesys.py [hunk 1]
```diff
@@ -187,17 +176,21 @@ async def generate_post():
...
-                    cnt = 0
+                        elif method == "GET":
+                            async with session.get(
+                                url,
+                                headers=self.authorization_token,
+                                params=params,
+                            ) as resp:
+                                new_report = await resp.read()
+                                semaphore.release()
+                await asyncio.sleep(0.5)
```
Comment (eSlider) [inline viadot/sources/genesys.py:187]: Does it need to be documented?
Maybe better to bring timeout-setting into arguments?

Comment (Diego-H-S) [reply viadot/sources/genesys.py:187]: What do you mean by documented?
The timeout parameter is passed at the task level, where do you suggest to put into the code?

Comment (eSlider) [reply viadot/sources/genesys.py:187]: it is good to give it as an argument to a method and describing as other arguments.

Comment (eSlider) [reply viadot/sources/genesys.py:187]: I mean `sleep_timeout = 0.5` as an argument.

**Review by Diego-H-S — COMMENTED** [review]
new `sleep_time` parameter implemented.

**Review by Rafalz13 — COMMENTED** [review]
Looks good.
Please add or update tests for flow/task.

**Review by Rafalz13 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,10 +5,15 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 ## [Unreleased]
 ### Added
+- Added a new `end_point` parameter in `genesys_api_connection` to make it more generic.
 
 ### Fixed
 
 ### Changed
+- Genesys API call method and the name changed from `genesys_generate_exports` to `genesys_api_connection`. 
+- Removed methods never used in production: `get_analitics_url_report`, `get_all_schedules_job`, `schedule_report`,
+`to_df`, `delete_scheduled_report_job` and `generate_reporting_export`.
```
[the remark this answers is not in the document]
Comment (Diego-H-S) [reply CHANGELOG.md:18]: done!

**Pull request merged**